support-frontend/assets/pages/contributions-landing/contributionsLandingInit.ts

```diff
@@ -167,32 +164,47 @@ function selectInitialAmounts(
 ) {
 	const { amounts } = state.common;
 	const amountFromUrl = getAmountFromUrl();
-	Object.keys(amounts).forEach((contributionType) => {
+
+	const amountForContributionType = (contributionType: ContributionType) => {
 		if (amountFromUrl && contributionType === selectedContributionType) {
 			if (
 				amountFromUrl !== 'other' &&
 				amounts[contributionType].amounts.includes(amountFromUrl)
 			) {
-				dispatch(selectAmount(amountFromUrl, contributionType));
-			} else {
-				dispatch(selectAmount('other', contributionType));
-
-				if (amountFromUrl !== 'other') {
-					dispatch(updateOtherAmount(`${amountFromUrl}`, contributionType));
-				}
+				return amountFromUrl;
 			}
-		} else {
-			const { defaultAmount } = amounts[contributionType];
-			dispatch(selectAmount(defaultAmount, contributionType));
+			return 'other';
 		}
-	});
+		return amounts[contributionType].defaultAmount;
```

Comment:
This is not something to stop this PR, but I was trying to understand the wider code and thinking about it, and my comment might be useful:
this code does feel a bit upside down (not directly from your changes, I think just the evolution)
I think what it's trying to do is return the default amount for all other than the "selected-contribution-type" from the url, and for the one that does match, it would read the amount from the url.
I feel like it might be clearer if this just mapped through all the `amounts` array to either pick out the defaultAmount, or if the selected type, then get from the url.
Would it even make sense to just dispatch that object, and then the receiving end could decide whether that is in the amounts for that contribution type?  I'm not sufficiently up on design patterns to know if that makes sense.

Reply:
Okay I had a go at addressing this one and the other comment about duplicating logic and have just pushed up a change. Let me know what you think!